relay-server/src/services/processor/event.rs

```diff
@@ -602,7 +602,13 @@ fn extract_attached_event(
     // Protect against blowing up during deserialization. Attachments can have a significantly
     // larger size than regular events and may cause significant processing delays.
     if item.len() > config.max_event_size() {
-        return Err(ProcessingError::PayloadTooLarge(item.ty().to_owned()));
+        return Err(ProcessingError::PayloadTooLarge(
+            if let Some(attachment_type) = item.attachment_type() {
+                attachment_type.clone().into()
+            } else {
+                item.ty().clone().into()
+            },
```

Comment:
The `clone`s are unnecessary. You could also do this with `map`, but I'm not sure if it's better.
```suggestion
            item.attachment_type()
                .map(|t| t.into())
                .unwrap_or_else(|| item.ty().into()),
```